graylog2-server/src/main/java/org/graylog/plugins/map/config/GeoIpResolverConfig.java

```diff
@@ -42,29 +51,50 @@ public abstract class GeoIpResolverConfig {
     @JsonProperty("asn_db_path")
     public abstract String asnDbPath();
 
+    @JsonProperty(FIELD_REFRESH_INTERVAL_UNIT)
+    @Nullable
+    public abstract TimeUnit refreshIntervalUnit();
+
+    @JsonProperty(FIELD_REFRESH_INTERVAL)
+    @Min(0)
+    public abstract long refreshInterval();
+
+    public Duration refreshIntervalAsDuration() {
+        if (refreshIntervalUnit() == null) {
+            return Duration.ofMinutes(DEFAULT_INTERVAL_MINUTES);
+        }
+        return Duration.ofMillis(refreshIntervalUnit().toMillis(refreshInterval()));
+    }
+
     @JsonCreator
     public static GeoIpResolverConfig create(@JsonProperty("enabled") boolean cityEnabled,
                                              @JsonProperty("enforce_graylog_schema") boolean enforceGraylogSchema,
                                              @JsonProperty("db_vendor_type") DatabaseVendorType databaseVendorType,
                                              @JsonProperty("city_db_path") String cityDbPath,
-                                             @JsonProperty("asn_db_path") String asnDbPath) {
+                                             @JsonProperty("asn_db_path") String asnDbPath,
+                                             @JsonProperty(FIELD_REFRESH_INTERVAL_UNIT) TimeUnit refreshIntervalUnit,
+                                             @JsonProperty(FIELD_REFRESH_INTERVAL) long refreshInterval) {
         return builder()
                 .enabled(cityEnabled)
                 .enforceGraylogSchema(enforceGraylogSchema)
                 .databaseVendorType(databaseVendorType == null ? DatabaseVendorType.MAXMIND : databaseVendorType)
                 .cityDbPath(cityDbPath)
                 .asnDbPath(asnDbPath)
+                .refreshIntervalUnit(refreshIntervalUnit)
+                .refreshInterval(refreshInterval)
```

Comment:
Updated.  I assume you meant:

```Java
 .refreshIntervalUnit(refreshIntervalUnit == null ? TimeUnit.MINUTES : refreshIntervalUnit)
  .refreshInterval(refreshInterval == null ? DEFAULT_INTERVAL_MINUTES : refreshInterval)
```
That is, setting to the actual argument if not null, rather than null.